cql3/statements/create_table_statement.cc

```diff
@@ -96,10 +97,30 @@ std::vector<column_definition> create_table_statement::get_columns()
     return column_defs;
 }
 
+template <typename CreateTable>
+future<shared_ptr<cql_transport::event::schema_change>>
+create_table_statement::create_table_with_cdc(service::storage_proxy& proxy,
+                                              schema_ptr schema,
+                                              CreateTable&& create_table) {
+    if (_if_not_exists) {
+        throw exceptions::invalid_request_exception(
+                "Can't create table with CDC support using IF NOT EXISTS");
+    }
+    cdc::db_context ctx = cdc::db_context::builder(proxy).build();
+    return cdc::setup(ctx, schema).then([ctx, create_table = std::move(create_table), schema = std::move(schema)] () mutable{
+        return create_table().handle_exception([ctx, schema = std::move(schema)](std::exception_ptr ep) mutable {
+            return cdc::remove(ctx, schema->ks_name(), schema->cf_name()).then([ep = std::move(ep)] () -> shared_ptr<cql_transport::event::schema_change> {
+                std::rethrow_exception(ep);
+            });
+        });
+    });
+}
+
 future<shared_ptr<cql_transport::event::schema_change>> create_table_statement::announce_migration(service::storage_proxy& proxy, bool is_local_only) {
-    auto create_table = [this, is_local_only, &proxy] {
-        return make_ready_future<>().then([this, is_local_only, &proxy] {
-            return service::get_local_migration_manager().announce_new_column_family(get_cf_meta_data(proxy.get_db().local()), is_local_only);
+    auto schema = get_cf_meta_data(proxy.get_db().local());
+    auto create_table = [this, is_local_only, schema] () mutable {
```

Comment:
I'm having a sense of deja-vu - haven't I seen this lambda in a previous patch?

Reply:
Exactly. It was prepared in a previous patch. Here I only change &proxy to schema and add mutable to the lambda.